pkg/broker/filter/filter_handler.go

```diff
@@ -210,6 +242,18 @@ func (h *Handler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 			return
 		}
 
+		oidcIdentity, err := getOIDCIdentity(request)
+		if err != nil {
+			h.logger.Warn("Error when extracting OIDC identity from the request", zap.Error(err))
+			return
+		}
+
+		if trigger.Status.Auth.ServiceAccountName != &oidcIdentity {
```

Comment:
We need to check based on the Subscriptions OIDC SA, not on the Triggers one. 
This is because for requests **to** the mt-broker-filter, the Subscriptions OIDC SA identity is used, but for requests **from** the mt-broker-filter, the triggers OIDC SA identity is used. As we want to check on the mt-broker-filter receivers side, we need to check that the Subscriptions OIDC SA was used.